dropshot/examples/file_server.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021 Oxide Computer Company
+
+use dropshot::ApiDescription;
+use dropshot::ConfigLogging;
+use dropshot::ConfigLoggingLevel;
+use dropshot::HttpError;
+use dropshot::HttpServerStarter;
+use dropshot::RequestContext;
+use dropshot::{endpoint, Path};
+use http::{Response, StatusCode};
+use hyper::Body;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+/**
+ * Our context is simply the root of the directory we want to serve.
+ */
+struct FileServerContext {
+    base: PathBuf,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), String> {
+    /*
+     * We must specify a configuration with a bind address.  We'll use 127.0.0.1
+     * since it's available and won't expose this server outside the host.  We
+     * request port 0, which allows the operating system to pick any available
+     * port.
+     */
+    let config_dropshot = Default::default();
+
+    /*
+     * For simplicity, we'll configure an "info"-level logger that writes to
+     * stderr assuming that it's a terminal.
+     */
+    let config_logging = ConfigLogging::StderrTerminal {
```

Comment:
Mostly because this is a convenient place for me to ask: suppose you wanted the logging level to be configurable via a command line argument. Have we picked a crate and style for that sort of thing?

Reply:
yeesh. Good question and not one I have an answer to. Seems like something @davepacheco or @smklein might know off the top of their heads.

Reply:
@kulakowski I've been accepting a flag usually named `--log-level` or similar, which is a `slog::Level`, and using that type's `FromStr` implementation. This is basically how the [`nexus` binary works](https://github.com/oxidecomputer/omicron/blob/3a6531e46d24f8efa3eeb57fc866d621acd9eb9a/omicron-nexus/src/config.rs#L18) as well, although that's reading from a TOML file instead of the command-line. Hope that helps!